dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -63,13 +71,14 @@ async def _setup_runtime_env(serialized_runtime_env, session_dir):
                 "Runtime env creation failed for %d times, "
                 "don't retry any more.",
                 runtime_env_consts.RUNTIME_ENV_RETRY_TIMES)
```

Comment:
Oh do we already retry above?

Reply:
Yep that's right, we retry creating the env 3 times by default before ever marking it failed. My change here is just caching the failure once we've already failed 3x so we don't go through the whole loop again. Effectively once an env fails 3x in a row, we assume it will never succeed in the future. We may want something more sophisticated in the future if we run into transient failures but this seems reasonable & simple to me for now.